ext/nnstreamer/tensor_decoder/tensordec-flatbuf.cc

```diff
@@ -0,0 +1,160 @@
+/**
+ * GStreamer / NNStreamer tensor_decoder subplugin, "Flatbuffer"
+ * Copyright (C) 2020 Gichan Jang <gichan2.jang@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file        tensordec-flatbuf.cc
+ * @date        26 Fev 2020
+ * @brief       NNStreamer tensor-decoder subplugin, "flatbuffer",
+ *              which converts tensor or tensors to flatbuffer byte stream.
+ *
+ * @see         https://github.com/nnsuite/nnstreamer
+ * @author      Gichan Jang <gichan2.jang@samsung.com>
+ * @bug         No known bugs except for NYI items
+ *
+ */
+
+#include <glib.h>
+#include <gst/gstinfo.h>
+#include <nnstreamer_plugin_api_decoder.h>
+#include <nnstreamer_plugin_api.h>
+#include "nnstreamer_generated.h" // Generated by `flatc`.
+
+using namespace NNStreamer;
+
+void init_fb (void) __attribute__ ((constructor));
+void fini_fb (void) __attribute__ ((destructor));
+
+#define DECODER_FB_BIN_CAPS_STR "application/octet-stream"
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+fb_init (void **pdata)
+{
+  *pdata = NULL;
+  return TRUE;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static void
+fb_exit (void **pdata)
+{
+  return;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+fb_setOption (void **pdata, int opNum, const char *param)
+{
+  return TRUE;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static GstCaps *
+fb_getOutCaps (void **pdata, const GstTensorsConfig * config)
+{
+  return gst_caps_from_string (DECODER_FB_BIN_CAPS_STR);
+}
+
+static GstFlowReturn
+fb_decode (void **pdata, const GstTensorsConfig * config,
+    const GstTensorMemory * input, GstBuffer * outbuf)
+{
+  GstMapInfo out_info;
+  GstMemory *out_mem;
+  flatbuffers::FlatBufferBuilder builder;
+  std::vector<flatbuffers::Offset<Tensor>> tensor_vector;
+  flatbuffers::Offset<flatbuffers::String> tensor_name;
+  int num_tensors = config->info.num_tensors;
+  auto fr = frame_rate(config->rate_n, config->rate_d);
+
+  /* Fill the info in tensor and puth to tensor vector */
+  for(int i = 0; i<num_tensors; i++) {
```

Comment:
i < num_tensors

And if the type of num_tensors is uint, you are trying to compare signed with unsigned.

Reply:
Updated.